### ISSUE TITLE
aria-conditional-attr should flag checkboxes even if the states are aligned

### ISSUE THREAD

**Issue by akronfeldTestParty2314** [dequelabs/axe-core#4810]
### Product

axe-core

### Feature Description

Currently axe-core only flags aria-conditional-attr on a checkbox if aria-checked and the native state are out of sync IE `<input type="checkbox" id="cb2" aria-checked="'false'" checked/>`. This defeats the purpose.  Since WCAG says the practice is an anti-pattern in general, it would be ideal to flag this even when the states are aligned. https://dequeuniversity.com/rules/axe/4.10/aria-conditional-attr?application=AxeChrome

**Comment by akronfeldTestParty2314** [comment]
> [@akronfeldTestParty2314](https://github.com/akronfeldTestParty2314) Thanks for the suggestion. I agree with you that this is bad code, but there is no accessibility issue as long as the two are kept in sync. The `aria-conditional-attr` is a WCAG rule, so it shouldn't fail things unless they are guaranteed WCAG failures. I think it would make sense to have this under a best practice rule. I don't think it fits under any of the existing BP rules, so it would have to be a new one. Would TestParty be interested in contributing such a rule?

Hi @WilcoFiers TestParty would be delighted to make that contribution yes!  I will get started on that right away.
